File: 747_Largest_Number_At_Least_Twice_of_Others.js
var dominantIndex = function(nums) {
    let x = Math.max(...nums);
    let index_x = nums.indexOf(x);
    nums[index_x] = -1;
    if(x >= Math.max(...nums)*2){
        return index_x;
    }
    else{
        return -1;
    } 
};

File: 747_Largest_Number_At_Least_Twice_of_Others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        int ans, mx = 0, secondMax = 0, n = nums.size();
        for(int i = 0; i < n; i++){
            if(nums[i] > mx){
                secondMax = mx;
                mx = nums[i];
                ans = i;
            } else if(nums[i] > secondMax){
                secondMax = nums[i];
            }
        }
        return secondMax > mx / 2 ? -1 : ans;
    }
};
